code/bart_stability.R

```diff
@@ -54,6 +54,35 @@ get_data <- function(dgp,n, q, rho=0.05, seed=0){
 
 }
 
+# Assumes that the dgp argument is in c("2016","2017","2018","2019")
+# Throws an error if the n + 200 > the number of rows in the data
+get_data_acic <- function(dgp,n){
+
+  train_n <- n
+  test_n <- 200
+
+  if (!(dgp %in% c("2016","2017","2018","2019"))) {
+    stop("Not a supported dgp")
+  }
+
+  data <- read.csv(paste0("data/ACIC",dgp,".csv"))
+
+  if (nrow(data) < train_n+test_n) {
+    stop("N is too large for the current dgp")
+  }
+
+  data_train <- data[1:train_n,-which(colnames(data) == "Ytrue")]
```

Comment:
@theo-s why not take the test set as a random? Looking at the data it seems like similar outcomes are clustered together